# Overview of stMixer
stMixer for scalable mosaic integration and label transfer in spatial histology and multi-omics
![](https://github.com/YQX-code/stMixer/blob/main/stMixer/fig1.png)
**a** *stMixer* supports the integration of diverse modalities (including H&E and omics), multi-slide mosaic integration, and label transfer anchored by either histology or omics.**b** For each input slide with spatial multi-modal data, *stMixer* employs a multi-view graph self-supervision module to learn harmony-preserving yet modality-specific representations (top), followed by a self-looped cross-attention mechanism that iteratively fuses these features into a unified embedding $Z_i^{O}$ (bottom).**c** Given multiple slide-specific embeddings as input, *stMixer* jointly optimizes triplet and MMD losses to position biologically similar cells nearby within a shared latent space.**d** For label transfer, *stMixer* introduces a two-stage, graph-enabled cluster-level voting strategy that propagates annotations from reference profiles to query slides.**e** *stMixer* enables flexible integration for spatial domain identification and inter-domain relations characterization, scalable multi-slide mosaic omics alignment, and cross-slide label transfer via soft link.
# Requirements
- python==3.8.20
- torch==2.4.1+cu121
- torchvision==0.19.1
- torchaudio==2.4.1
- torch-geometric==2.6.1
- torch_cluster==1.6.3
- torch_scatter==2.1.2
- cudnn==90100
- numpy==1.24.4
- scanpy==1.9.8
- anndata==0.9.2
- pandas==2.0.3
- scipy==1.10.1
- scikit-learn==1.3.2
- scikit-misc==0.2.0
- tqdm==4.67.0
- matplotlib==3.6.3
- matplotlib-inline==0.1.7
- harmony-pytorch==0.1.8


# Data
stMixer was applied to both synthetic benchmarks and nine real-world spatial datasets generated by five different technologies, covering lymph node, thymus, brain, spleen, and breast cancer.The human lymph node dataset was profiled using the 10x Genomics platform and is available from SpatialGlue (Long et al. 2024).The mouse thymus dataset was generated using Stereo-CITE-seq (Liao et al. 2023, unpublished), and can be downloaded from the Spatial Transcript Omics DataBase(https://db.cngb.org/stomics/project/STT0000094). Three mouse brain datasets were profiled using the spatial ATAC-RNA-seq platform (Zhang et al. 2023), and have been deposited in the Gene Expression Omnibus under accession code GSE205055 (https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE205055). Two mouse spleen datasets were generated using the SPOTS platform (Ben-Chetrit et al. 2023), and have been deposited in the Gene Expression Omnibus under accession number GSE198353 (https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE198353). Two human breast cancer datasets are available on the 10x Genomics website (https://www.10xgenomics.com/datasets/). The mouse brain single-cell dataset (Zeisel et al. 2018) is deposited in the sequence read archive under accession code SRP135960 (https://www.ncbi.nlm.nih.gov/sra/SRP135960).
